src/components/ButtonMain.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  <button
+    @click="$emit('click')"
```

Comment:
it may be a good idea to emit original event here to be able to e.g. call `preventDefault` on it from parent perspective